snapcraft/internal/pluginhandler.py

```diff
@@ -804,27 +857,56 @@ def _migratable_filesets(fileset, srcdir):
     return snap_files, snap_dirs
 
 
-def _migrate_files(snap_files, snap_dirs, srcdir, dstdir, missing_ok=False,
-                   follow_symlinks=False, fixup_func=lambda *args: None):
-
+def _migrate_dirs(snap_dirs, srcdir, dstdir, step='stage'):
     for directory in snap_dirs:
-        src = os.path.join(srcdir, directory)
-        dst = os.path.join(dstdir, directory)
+        copy_tree = False
+        if type(directory) == tuple:
+            src = os.path.join(srcdir, directory[0])
+            dst = os.path.join(dstdir, directory[1])
+            copy_tree = True
+        else:
+            src = os.path.join(srcdir, directory)
+            dst = os.path.join(dstdir, directory)
 
         snapcraft.file_utils.create_similar_directory(src, dst)
+        if copy_tree:
+            file_utils.link_or_copy_tree(src, dst)
+
+
+def _can_skip(src, dst, missing_ok=False):
+    if os.path.isdir(src):
+        snapcraft.file_utils.create_similar_directory(
```

Comment:
This makes me think that the name of the function is not right.
can_skip sounds like something that returns true or false, and that's it. This will create a directory as a side-effect, that I wouldn't expect.